subsys/net/ip/net_context.c

```diff
@@ -49,6 +49,9 @@
 #define ACK_TIMEOUT K_SECONDS(1)
 #endif
 
+/* TODO: It should be 2 * MSL (Maximum segment lifetime) */
+#define TIMEWAIT_TIMEOUT MSEC(100)
+
```

Comment:
This is another issue discovered, indeed. Will 100 ms be enough for the current cases or should the timeout be longer? 100 ms sounds awfully short for some reason.